src/Compilers/CSharp/Portable/Symbols/Metadata/PE/MemberRefMetadataDecoder.cs

```diff
@@ -153,16 +152,21 @@ internal Symbol FindMember(MemberReferenceHandle memberRef, bool methodsOnly)
             }
         }
 
-        private static FieldSymbol FindFieldBySignature(TypeSymbol targetTypeSymbol, string targetMemberName, ImmutableArray<ModifierInfo<TypeSymbol>> customModifiers, TypeSymbol type)
+        private static FieldSymbol FindFieldBySignature(TypeSymbol targetTypeSymbol, string targetMemberName, in FieldInfo<TypeSymbol> fieldInfo)
         {
             foreach (Symbol member in targetTypeSymbol.GetMembers(targetMemberName))
             {
                 var field = member as FieldSymbol;
                 TypeWithAnnotations fieldType;
 
+                // Ensure the field symbol matches the { IsByRef, RefCustomModifiers, Type, CustomModifiers }
+                // from metadata. The checks are not strictly necessary though since PENamedTypeSymbol
+                // does not support fields with duplicate names. Matching by name is sufficient.
```

Comment:
>Matching by name is sufficient. [](http://example.com/codeflow?start=65&length=31)

This sounds suspicious too. Matching name doesn't guarantee that the signature is matching, even when there is only one field with the name in the type.